### PR TITLE
MODNOTES-233 Do not include # of notes assigned to note types in response

### PR DESCRIPTION
**Pull request by maryna-steshenko**
## Purpose
Improve mod-notes performance by not returning a total number of notes assigned to a note type. 

## Approach
- Instead of a total number a value indicating whether a note type is assigned to any note is returned
- Corresponding changes to unit tests are made
- Karate tests: https://github.com/folio-org/folio-integration-tests/pull/652

### CODE REVIEW

File: src/main/java/org/folio/notes/domain/repository/NoteTypesRepository.java [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 public interface NoteTypesRepository extends JpaCqlRepository<NoteTypeEntity, UUID> {
 
-  @Query("SELECT n.type.id as typeId, COUNT(n.id) as count FROM NoteEntity AS n WHERE n.type.id IN (:noteTypeIds) GROUP BY n.type.id")
+  @Query("SELECT t.id as typeId, EXISTS (SELECT n.type.id FROM NoteEntity AS n WHERE n.type.id = t.id) as isAssigned FROM NoteTypeEntity as t WHERE t.id IN (:noteTypeIds)")
   List<NoteTypeCount> findNoteUsage(@Param("noteTypeIds") List<UUID> noteTypeIds);
```
Comment (psmagin) [inline src/main/java/org/folio/notes/domain/repository/NoteTypesRepository.java:16]:
```suggestion
  List<NoteTypeCount> findNoteUsage();
```
Let's remove these ids and query for all note types (we have programmatic limitations for the number of note types). I think it simplifies queries and will improve its performance.

File: src/main/java/org/folio/notes/domain/repository/NoteTypesRepository.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 public interface NoteTypesRepository extends JpaCqlRepository<NoteTypeEntity, UUID> {
 
-  @Query("SELECT n.type.id as typeId, COUNT(n.id) as count FROM NoteEntity AS n WHERE n.type.id IN (:noteTypeIds) GROUP BY n.type.id")
+  @Query("SELECT t.id as typeId, EXISTS (SELECT n.type.id FROM NoteEntity AS n WHERE n.type.id = t.id) as isAssigned FROM NoteTypeEntity as t WHERE t.id IN (:noteTypeIds)")
```
Comment (psmagin) [inline src/main/java/org/folio/notes/domain/repository/NoteTypesRepository.java:15]:
```suggestion
  @Query("SELECT t.id as typeId, EXISTS (SELECT n.type.id FROM NoteEntity AS n WHERE n.type.id = t.id LIMIT 1) as isAssigned FROM NoteTypeEntity as t WHERE t.id IN (:noteTypeIds)")
```
It's enough to have at least one record  in the subquery

File: src/main/resources/swagger.api/schemas/noteType.yaml [hunk 3]
```diff
@@ -15,10 +15,10 @@ noteType:
       readOnly: true
       description: Type usage statistics
       properties:
-        noteTotal:
-          type: integer
+        isAssigned:
+          type: boolean
           readOnly: true
-          description: Total number of notes of this type
+          description: Is this note type assigned to any note
```
Comment (psmagin) [inline src/main/resources/swagger.api/schemas/noteType.yaml:21]: It's a breaking change of the API. The interface version should be increased.

**Review by psmagin — CHANGES_REQUESTED** [review]

**Comment by psmagin** [comment]
This PR couldn't be merged until UI changes will be applied and all dependent modules change the dependent interface version.

**Review by psmagin — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json [hunk 4]
```diff
@@ -14,7 +14,7 @@
   "provides": [
     {
       "id": "notes",
-      "version": "2.0",
+      "version": "3.0",
```
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json:17]: This version change should be reflected in the YAML file describing the interface

(it is currently version 1.0, which suggests to me that the previous breaking change also did not change this)

Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json:17]: As the APIs provided by this module are now incompatible with the APIs provided by previous versions, this is a major implementation breaking change.

That should be reflected in a major version change for the module.

**Comment by zburke** [comment]
@maryna-steshenko, what is the timeline for merging this change? I see [MODNOTES-233](https://issues.folio.org/browse/MODNOTES-233) is targeted for Nolana. This not only affects ui-notes but also stripes-smart-components, which potentially makes this a breaking change for all of stripes and thus for every single UI application. 

Depending on what SSC does (or ignores) with the mod-notes API responses, we maybe able to change support from `"2"` to `"2 || 3"` and avoid the breaking change in stripes, but we need some time to investigate. @BogdanDenis, I see your name attached to some SSC commits related to `lib/Notes`. Is this something you could potentially investigate, or do you know who could?

**Comment by BogdanDenis** [comment]
> @maryna-steshenko, what is the timeline for merging this change? I see [MODNOTES-233](https://issues.folio.org/browse/MODNOTES-233) is targeted for Nolana. This not only affects ui-notes but also stripes-smart-components, which potentially makes this a breaking change for all of stripes and thus for every single UI application.
> 
> Depending on what SSC does (or ignores) with the mod-notes API responses, we maybe able to change support from `"2"` to `"2 || 3"` and avoid the breaking change in stripes, but we need some time to investigate. @BogdanDenis, I see your name attached to some SSC commits related to `lib/Notes`. Is this something you could potentially investigate, or do you know who could?

@zburke this change will only affect note types, and we have a ticket ([UINOTES-132](https://issues.folio.org/browse/UINOTES-132)) to account for this, we'll change API version there
But for SSC interface version 2 or 3 shouldn't matter because Notes don't use `/note-types` endpoint that was changed or the `usage.noteTotal` property that was removed - is that correct, @maryna-steshenko?
I'll create additional tickets to clean up things like test stubs and prop types in SSC and other modules.

**Pull request merged**